src/dev_cron.erl

```diff
@@ -17,11 +19,165 @@ info(_Msg1, _Msg2, _Opts) ->
+        <<"execution-device">> => <<"lua@5.3a">>,
+        <<"function">> => <<"compute">>,
+        <<"module">> => #{
+            <<"content-type">> => <<"application/lua">>,
+            <<"body">> => Script
+        }
+    },
+    ?event({debug_cron_opts, {cron_process_def, CronProcDef}}),
+    Opts#{
+        node_processes =>
+            (hb_opts:get(node_processes, #{}, Opts))#{
+                <<"cron">> => CronProcDef
+            }
+    }.
+
+%% @doc Normalize the cron jobs from cache by attempting to restart them.
+normalize(Msg1, _Msg2, Opts) ->
+	?event({normalize_cron_jobs_start}),
+    CronOpts = cron_opts(Opts),
+    case cache_list(CronOpts) of
+		{ok, Crons} -> 
+			?event({normalize_cron_jobs_loaded, {count, length(Crons)}}),
+            {Successes, Errors} = process_cached_jobs(Crons, Msg1, CronOpts),
+            NumSuccess = length(Successes),
+            ErrorDetails = Errors,
+            ?event({normalize_cron_jobs_finished, 
+				{success_or_existing, NumSuccess}, 
+				{errors, length(ErrorDetails)}, 
+				{error_details, ErrorDetails}}),
+			{ok, #{
+				<<"restarted_or_existing">> => NumSuccess, 
+				<<"errors">> => ErrorDetails
+			}};
+		{error, Reason} ->
+			 ?event({normalize_cron_jobs_load_error, {error, Reason}}),
+             {error, Reason}
+	end.
+
+%% Helper function to process the list of cached cron jobs
+process_cached_jobs(Crons, Msg1, CronOpts) ->
+    Results = lists:map(
+        fun(Job) -> 
+            process_single_job(Job, Msg1, CronOpts)
+        end, Crons
+    ),
+    lists:partition(
+		fun({ok, _}) -> true; 
+		(_) -> false end, 
+		Results).
+
+%% Helper function to process a single cached job entry
+process_single_job(Job, Msg1, CronOpts) ->
+    case extract_job_details(Job) of
+        {ok, DetailsMap} ->
+            case reconstruct_original_msg(DetailsMap) of
+                {ok, OriginalMsg} ->
+                    attempt_job_restart(DetailsMap, OriginalMsg, Msg1, CronOpts);
+                {error, Reason} ->
+                    TaskId = maps:get(task_id, DetailsMap, unknown),
+                    {error, {TaskId, Reason}}
+            end;
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+%% Helper function to extract and validate details from a raw cache job entry
+extract_job_details(Job) ->
+    Body = maps:get(<<"body">>, Job, #{}),
+    TaskId = maps:get(<<"task_id">>, Body, undefined),
+    AugmentedData = maps:get(<<"data">>, Body, #{}),
+    Type = maps:get(<<"type">>, AugmentedData, undefined),
+    Interval = maps:get(<<"interval">>, AugmentedData, undefined),
+    WorkerMsg = maps:get(<<"worker_msg">>, AugmentedData, #{}),
+    if TaskId == undefined orelse Type == undefined orelse not is_map(WorkerMsg) orelse WorkerMsg == #{} ->
+        ?event({normalize_skipping_invalid_job, {reason, invalid_structure}, {raw_job, Job}}),
+        {error, {unknown_task_id, invalid_job_structure}};
+       true ->
```

Comment:
Suggestion: In all the code base the map keys for messages are stringified and not atom based, so to be more consistent string key should used instead. So no need to figure it if we should use atom or string key.